modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -331,42 +380,83 @@ class ACASThingBrowserController extends Backbone.View
 			model: thingModel
 			query: @query
 			configs: @configs
+			maxResults: @maxResults
 			el: @$('.bv_thingSearchController')
 		@searchController.render()
 		@searchController.on "searchReturned", @setupThingSummaryTable.bind(@)
+		@searchController.on "find", @searchingStarted.bind(@)
+		@searchController.on "moreSpecificSearchRequired", @moreSpecificSearchRequired.bind(@)
 		@$('.bv_queryToolDisplayName').html window.conf.service.result.viewer.displayName
 
-	setupThingSummaryTable: (things) =>
+	showOne: (cl) =>
+		cls = ["bv_noMatchingThingsFoundMessage", "bv_moreSpecificThingSearchNeeded", "bv_errorOccurredPerformingSearch", "bv_thingBrowserSearchInstructions", "bv_searchingThingsMessage"]
+		for c in cls
+			@$("." + c).addClass("hide")
+		if cl?
+			@$("." + cl).removeClass("hide")
+		@$(".bv_thingTableController").addClass("hide")
+		@$(".bv_searchThingsStatusIndicator").removeClass("hide")
+
+	moreSpecificSearchRequired: =>
+		@showOne("bv_moreSpecificThingSearchNeeded")
+
+	searchingStarted: =>
 		@destroyThingSummaryTable()
+		@showOne("bv_searchingThingsMessage")
 
-		@$(".bv_searchingThingsMessage").addClass "hide"
-		if things is null
-			@$(".bv_errorOccurredPerformingSearch").removeClass "hide"
+	setupThingSummaryTable: (response) =>
+		results = response.results
 
-		else if things.length is 0
-			@$(".bv_noMatchingThingsFoundMessage").removeClass "hide"
-			@$(".bv_thingTableController").html ""
+		if response.maxResults <= response.numberOfResults
+			msg = "Browser results were limited to the first " + response.maxResults + " entries out of " + response.numberOfResults + " results"
+			@$('.bv_maxThingBrowserSearchResultsReached').html msg
+			@$(".bv_maxThingBrowserSearchResultsReached").removeClass("hide")
+		else 
+			@$(".bv_maxThingBrowserSearchResultsReached").addClass("hide")
+
+		if results is null
+			@showOne("bv_errorOccurredPerformingSearch")
+
+		else if results.length is 0
+			@showOne("bv_noMatchingThingsFoundMessage")
 		else
 			@$(".bv_searchThingsStatusIndicator").addClass "hide"
-			@$(".bv_thingTableController").removeClass "hide"
-			thingCollection =  Backbone.Collection.extend 
-				model: @modelClass
+			@$(".bv_thingTableController").removeClass "hide" 
+
 			@thingSummaryTable = new ThingSummaryTableController
-				collection: new thingCollection things
+				collection: results
+				thingModelClass: @modelClass
 				configs: @configs
 				columnFilters: @columnFilters
+				maxResults: @maxResults
 
 			@thingSummaryTable.on "selectedRowUpdated", @selectedThingUpdated
 			@$(".bv_thingTableController").html @thingSummaryTable.render().el
 
 	selectedThingUpdated: (thing) =>
 		@$('.bv_thingControllerWrapper').append("<div class='bv_thingController'></div>")
 		@trigger "selectedThingUpdated"
-		@thingController = new @controllerClass
-			el: @$('.bv_thingController')
-			model: thing
-			readOnly: true
```

Comment:
I don't see `readOnly` in your new split implementation below. I think that is important to pass, right?

Reply:
Thanks.  I did not remove this on purpose. It's strange it's hardcoded to true but I'm not fixing that here.

Reply:
Fixed by adding it back